test/integration/scheduler/podsready/scheduler_test.go

```diff
@@ -276,6 +276,27 @@ var _ = ginkgo.Describe("SchedulerWithWaitForPodsReady", func() {
 				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl), prodWl)).Should(gomega.Succeed())
 				g.Expect(ptr.Deref(prodWl.Spec.Active, true)).Should(gomega.BeFalse())
 			}, util.Timeout, util.Interval).Should(gomega.Succeed())
+
+			ginkgo.By("verify the re-activated inactive 'prod' workload re-queue state is reset")
+			gomega.Eventually(func(g gomega.Gomega) {
+				g.Expect(k8sClient.Get(ctx, client.ObjectKeyFromObject(prodWl), prodWl)).Should(gomega.Succeed())
+				apimeta.SetStatusCondition(&prodWl.Status.Conditions, metav1.Condition{
+					Type:    kueue.WorkloadEvicted,
+					Status:  metav1.ConditionTrue,
+					Reason:  kueue.WorkloadEvictedByDeactivation,
+					Message: "evicted by Test",
+				})
+				g.Expect(k8sClient.Status().Update(ctx, prodWl)).Should(gomega.Succeed())
```

Comment:
wouldn't the controller add this already?

Reply:
Although the jobs controller adds this condition here: https://github.com/kubernetes-sigs/kueue/blob/a96927de470d985f17b03a328c33bb8547eae9a4/pkg/controller/jobframework/reconciler.go#L423, we don't register the jobs controller to the manager. 

https://github.com/kubernetes-sigs/kueue/blob/975b5f238ee08c27c5988ec46a52ce1f29e38285/test/integration/scheduler/podsready/suite_test.go#L68-L104

Reply:
We just register the indexers: https://github.com/kubernetes-sigs/kueue/blob/975b5f238ee08c27c5988ec46a52ce1f29e38285/test/integration/scheduler/podsready/suite_test.go#L96-L97

Reply:
Interesting...
We should probably move that piece of code to the Workload controller in a follow up. It doesn't belong in the job reconciler, as it is completely independent from the job. Otherwise, if anyone wants to write a custom integration, they would need to implement this.

Reply:
It makes sense, but I need to remember why we put this here.
Let me open an issue for this.

Reply:
Can you add a TODO to remove this status update and link to https://github.com/kubernetes-sigs/kueue/issues/1841?

Reply:
Sure